### DISCUSSION TITLE
"Geo multi key index is not supported" error while saving item, running on AWS documentDB

### DISCUSSION THREAD

**Discussion by mvdve** [payloadcms/payload#2605] [Q&A]
Hi All,

We are completely stuck with our deployment to production of our backend. The app is running fine within app runner and can connect to an AWS documentDB instance, but when i save an item in one specific collection, i get the following error:

"Geo multi key index is not supported".

This collection has some basic text, image and relationship fields and one field with a lot of blocks. which are not added.

Does somebody has a suggestion on what the cause can be and how to solve this? Thanks.

**Comment by mvdve** [comment]
For now replaced the point field with two number fields as a work around.

**Reply by DanRibbens** [reply]
Looks like documentDB does not support `near` queries. https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html#mongo-apis-geospatial

Today I learned!

Moving to number fields was a good workaround. I would expect you to be able to have the point field, as long as you aren't querying on it. Is the server not starting for you at all?

**Reply by DanRibbens** [reply]
Payload's point querying using `$near` but it looks like `$nearSphere` could be used and is supported by documentDB. I had to look up the difference, flat projection distance vs spherical. In most cases people probably want $nearSphere over the current implemented way.

I think we need to add $nearSphere as an additional query operator, which is probably what most people are using the point field for.

**Reply by mvdve** [reply]
Hi Dan, thanks for verifying this so quickly. The server does start without issues, but we could not save any content within this collection. The collection itself did not have a point field but a block used in the collection did. 

The system always tries to load all blocks within the GraphQL query with an inline fragment. Something like

```
    ... on ContactMap {
        blockType
        title
        address
        gps
    }
```

**Reply by DanRibbens** [reply]
It is surprising to me that you get an error on save. The index is supposed to be compatible so it must be something with the point data shape or type instead.

This will need more investigation.
